### ISSUE TITLE
discarded_notification_center_observer false trigger

### ISSUE THREAD

**Issue by freak4pc** [realm/SwiftLint#2442]
### New Issue Checklist

**If you approve this is a real issue, I can easily make a PR for this. Just looking for confirmation that this needs fixing**

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`discarded_notification_center_observer ` is being incorrectly triggered for iOS >= 9, maxOS >= 10.11.

If you look into Apple's documentation on removeObserver: https://developer.apple.com/documentation/foundation/notificationcenter/1413994-removeobserver

You can see the following: 
> If your app targets iOS 9.0 and later or macOS 10.11 and later, you don't need to unregister an observer in its dealloc method. Otherwise, you should call this method or removeObserver(_:name:object:) before observer or any object specified in addObserver(forName:object:queue:using:) or addObserver(_:selector:name:object:) is deallocated.

Which seems to mean that the pointer doesn't necessarily need to be stored, as its removed on deallocated. 

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

> Discarded Notification Center Observer Violation: When registering for a notification using a block, the opaque observer that is returned should be stored so it can be removed later. (discarded_notification_center_observer)

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.27.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? CocoaPods

* Which Xcode version are you using (check `xcode-select -p`)? Xcode 10
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`

```swift
NotificationCenter.default.addObserver(forName: NSNotification.Name.UIKeyboardDidHide, object: nil, queue: nil) { notification in 

}
```

![image](https://user-images.githubusercontent.com/605076/47115203-81254280-d266-11e8-8882-00385ae04332.png)

**Comment by marcelofabri** [comment]
I don't think it's a bug, as you should still call `removeObserver(_:)` in that case.

There's a more detailed explanation here: https://oleb.net/blog/2018/01/notificationcenter-removeobserver/.

**Comment by freak4pc** [comment]
Wow,that is an odd documentation mix up. Appreciate the prompt response!
